lms/djangoapps/course_api/blocks/serializers.py

```diff
@@ -67,6 +72,23 @@ def to_representation(self, block_key):
                     supported_field.default_value,
                 )
                 if field_value is not None:
+
+                    # override due date with student extension in unit added by instructor.
+                    if supported_field.block_field_name is 'due':
```

Comment:
why we want to execute this condition only when `field_value` is already populated?

Reply:
that's the actual bug that is fixed in it :)